tests/app/views/test_suppliers.py

```diff
@@ -1427,89 +1520,226 @@ def test_countersigned_at_timestamp_cannot_be_set(self):
         agreement_details_payload = {
-            "some": [
-                "arbitrary",
-                123,
-                ["json"]
-            ],
-            "here": "there",
+            "signerName": "name",
+            "signerRole": "role",
         }
         response = self.supplier_framework_update(
-            0, 'digital-outcomes-and-specialists',
+            0, 'g-cloud-8',
             update={'agreementDetails': agreement_details_payload})
 
-        assert_equal(response.status_code, 200)
+        assert response.status_code == 200
         data = json.loads(response.get_data())
-        assert_equal(data['frameworkInterest']['agreementDetails'], agreement_details_payload)
+        assert data['frameworkInterest']['agreementDetails'] == agreement_details_payload
 
         # while we're at it let's test the agreementDetails partial updating behaviour
         agreement_details_update_payload = {
-            "other": {
-                "json": 456,
-            },
-            "here": None,
+            "uploaderUserId": 1,
         }
         response2 = self.supplier_framework_update(
-            0, 'digital-outcomes-and-specialists',
-            update={'agreementDetails': agreement_details_update_payload})
+            0, 'g-cloud-8',
+            update={
+                'agreementReturned': True,
+                'agreementDetails': agreement_details_update_payload
+            }
+        )
 
         agreement_details_payload.update(agreement_details_update_payload)
-        # json validator should strip this key
-        del agreement_details_payload["here"]
+        assert response2.status_code == 200
+        data2 = json.loads(response2.get_data())
+        assert data2['frameworkInterest']['agreementDetails'] == {
+            "signerName": "name",
+            "signerRole": "role",
+            "uploaderUserId": 1,
+            "uploaderUserName": "my name",
+            "uploaderUserEmail": "test+1@digital.gov.uk",
+            "frameworkAgreementVersion": "v1.0",
+        }
 
-        assert_equal(response2.status_code, 200)
+    def test_can_not_set_agreement_details_on_frameworks_without_framework_agreement_version(self):
+        agreement_details_payload = {
+            "signerName": "name",
+            "signerRole": "role",
+            "uploaderUserId": 1,
+        }
+        response = self.supplier_framework_update(
+            0, 'digital-outcomes-and-specialists',
+            update={'agreementDetails': agreement_details_payload})
+
+        assert response.status_code == 400
+        data = json.loads(response.get_data())
+        strings_we_expect_in_the_error_message = [
+            'Framework', 'digital-outcomes-and-specialists', 'does not accept',  'agreementDetails']
+        for error_string in strings_we_expect_in_the_error_message:
+            assert error_string in data['error']
+
+    def test_setting_agreement_details_with_nonexistent_user_id_doesnt_return_user_details(self):
```

Comment:
Nitpick... this test name isn't quite right any more... something along the lines of
`test_can_not_set_agreement_details_with_nonexistent_user_id`?